build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -311,6 +312,7 @@ fun configureTests() {
         }
 
         if (project.supportsPredictiveTestSelection() && !isUnitTest()) {
+            rootProject.extensions.findByType<BuildScanExtension>()?.tag("PTS")
```

Comment:
Shouldn't this be `enabled && !rerunAllTests` and only add the tag if PTS is on? Moreover, would we add the tag here multiple times, one for each test task?

One thing we could do here is to add a custom value per test task, like `pts_<task-path>` to capture which tests had PTS enabled. WDYT?

Reply:
> Shouldn't this be enabled && !rerunAllTests and only add the tag if PTS is on?

I hesitated about this. More precisely, should we add the tag only when the task is really executed (not skipped/up-to-date/from-cache)? And what if some other build script disable a Test task after this code?

But adding the custom value for each task is  a good idea.

Reply:
The problem is that in its current state, the PR would add the tag to any build which runs a task which supports PTS, not when PTS is actually in use. I think using `Project.predictiveTestSelectionEnabled` in a buildFinished hook (build scan build finished) and adding the tag based on that would already go a long way. WDYT?